### ISSUE TITLE
Error Handling

### ISSUE THREAD

**Issue by michaelrabie** [alphasights/ember-graphql-adapter#34]
The way we are handling GraphQL errors is causing issues when communicating on the front end and running respective tests.

@Arkham

**Comment by Arkham** [comment]
The issue here is caused by the graphql errors not referencing the field which is causing the error.

From the graphql spec here (https://facebook.github.io/graphql/#sec-Errors) it says that we can just attack our own fields. My proposal would be that each error should have a 'source' field which tells us which field is causing the error. In the case such field is empty we can assume that it is a generic error of the object.

So the API would send us back something like this:

```
{
  errors: [{
    source: "advisor_phone_number",
    message: "Argument 'advisor_phone_number' on Field 'interaction' has an invalid value",
    locations: [{"line":1,"column":31}]
  }]
}
```

This would require to change the graphql-ruby gem to include this information and change the extractErrors function in our graphql serializer.

Thoughts? @toreriklinnerud @daemonsy @josercruz01 @brianstorti

**Comment by toreriklinnerud** [comment]
@Arkham Yes sounds good to me, just wondering if it makes sense to treat validation errors as "type" errors, but I guess why not.

**Comment by toreriklinnerud** [comment]
we have to make sure that these errors aren't reported by grapql as actual errors (as I believe that the current behaviour is to raise an exception after the response has been sent) if there are errors and we want to retain that behaviour for non-validation errors

**Comment by toreriklinnerud** [comment]
https://github.com/alphasights/pistachio/pull/5348

**Comment by delkopiso** [comment]
Fixed by #36
